decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -476,8 +476,20 @@ def within_edit_time_limit?
         return true if draft?
         return true if component.settings.proposal_edit_time == "infinite"
 
-        limit = updated_at + component.settings.proposal_edit_before_minutes.minutes
-        Time.current < limit
+        edit_time = component.settings.edit_time
```

Comment:
```suggestion
        time_value, time_unit = component.settings.edit_time
```
You can simply do this